src/pybind/mgr/cephadm/module.py

```diff
@@ -1320,6 +1337,15 @@ def _create_daemon(self, daemon_type, daemon_id, host,
                      'mgr', 'profile crash'],
         })
 
+        # osd deployments needs an --osd-uuid arg
+        if daemon_type == 'osd':
+            if not osd_uuid_map:
+                osd_uuid_map = self.get_osd_uuid_map()
+            osd_uuid = osd_uuid_map.get(daemon_id)
```

Comment:
This raises a ValueErorr, if `daemon_id not in osd_uuid_map`, which makes the code below unreachable. 

```suggestion
            osd_uuid = osd_uuid_map.get(daemon_id, None)
```

In general, can we move this code block upwards to the redeploy case?